src/Core/Domain/Validation/RegexPattern.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Validation;
```

Comment:
I don't think that `Validation` should be part of `Domain`.

Reply:
But it is domain rules validation

Reply:
As I understand, `Domain` should represent important concepts of PrestaShop (ecommerce) that could be Product, Category, Payment, Manufacturer & etc. 

Each of these subdomains have their own constraints and it happens to have similar (or identical) constraints between each other, this is where code duplication occurs. 

However, this allows different subdomains to evolve independently, without affecting each other.

In this particular case, you extracted common constraints into new subdomain called `Validation` which does not have any conceptual meaning for PrestaShop (ecommerce), at least this is how I see it.

@matks wdyt?

Reply:
makes sense :thinking: . Do you offer just having the patterns in each Domain VO's where it's needed and don't mind the duplication as those VO's 'coincidentally' have the same business rules ?

Reply:
Not sure, maybe I'd put it in `Core/Validation`, let's see what @matks thinks.

Reply:
Argh :/ I'm afraid there's no easy answer

Strictly speaking, Sarunas is right about keeping subdomains independent from each other.

However we could also consider this class to be some Utils class that is used in domains, but could live outside of it. For example Doctrine traits like TimestampableTrait are _usable and useful_ behaviors that are then _added_ to real domain elements (the entities). So they are not part of a subdomain but they are used in multiple of them.

So I think we can allow this class to exist, like the src/Core/Domain/Exception folder exists because it's used everywhere. It's a util class. It could be put inside a dedicated git repository and provided on packagist as "a useful repository of regex patterns to valide standard strings in the e-commerce world"

It's a very blurry line 😢